scoring/src/main/scala/ai/h2o/sparkling/ml/models/H2OMOJOModel.scala

```diff
@@ -234,22 +239,14 @@ object H2OMOJOModel extends H2OMOJOReadable[H2OMOJOModel] with H2OMOJOLoader[H2O
 
   def createFromMojo(mojo: File, uid: String, settings: H2OMOJOSettings): H2OMOJOModel = {
     val mojoModel = Utils.getMojoModel(mojo)
-    val model = mojoModel match {
-      case (_: SharedTreeMojoModel | _: XGBoostMojoModel) if mojoModel.isSupervised =>
-        new H2OTreeBasedSupervisedMOJOModel(uid)
-      case m: SharedTreeMojoModel if !m.isSupervised =>
-        new H2OTreeBasedUnsupervisedMOJOModel(uid)
-      case m if m.isSupervised => new H2OSupervisedMOJOModel(uid)
-      case _ => new H2OUnsupervisedMOJOModel(uid)
-    }
-
+    val model = createSpecificMOJOModel(uid, mojoModel._algoName)
     model.setSpecificParams(mojoModel)
     model.setMojo(mojo)
     val modelJson = getModelJson(mojo)
     val (trainingMetrics, validationMetrics, crossValidationMetrics) = extractAllMetrics(modelJson)
     val modelDetails = getModelDetails(modelJson)
     val modelCategory = extractModelCategory(modelJson)
-    val trainingParams = extractParams(modelJson)
+    val trainingParams = extractParams(modelJson) // TODO: To be deprecated
```

Comment:
just question, H2O is deprecating the model json?

Reply:
Nope, this is just a comment for me to deprecate `trainingParams`. Since we have algorithm specific MOJO classes with a getter method for each parameter, `trainingParams` are not needed IMO. I have a plan to do the deprecation in a separate PR.

Reply:
Thanks for the explanation!